manager/controllers/app/policy_compiler_interface.go

```diff
@@ -75,7 +75,11 @@ func LookupPolicyDecisions(datasetID string, policyManager connectors.PolicyMana
 
 	var creds string
 	if appContext.Application.Spec.SecretRef != "" {
-		creds = utils.GetVaultAddress() + vault.PathForReadingKubeSecret(appContext.Application.Namespace, appContext.Application.Spec.SecretRef)
+		if !utils.IsVaultEnabled() {
+			appContext.Log.Error().Str("SecretRef", appContext.Application.Spec.SecretRef).Msg("SecretRef defined [%s], but vault is disabled")
+		} else {
+			creds = utils.GetVaultAddress() + vault.PathForReadingKubeSecret(appContext.Application.Namespace, appContext.Application.Spec.SecretRef)
+		}
```

Comment:
There is the assumption that if vault is disabled then SecretRef in fybrikapp should not be used. it makes sense to me but not sure if that would always be the case. for example, if the datacatalog could retrieve the creds from the secret without vault.

Reply:
Right. Instead of an error, you can pass namespace/name of the secret.

Reply:
I don't think it is a good idea to return `namespace/name `, at least now. 
The `creds` string is used as an URL to send grpc or OpenAPI requests. For K8s secret base certificates, we need to create an additional pass. Let's see if we need it.
Thoughts ?

Reply:
Ok, let's stay with an error.